packages/codemirror/src/searchprovider.ts

```diff
@@ -590,7 +431,8 @@ export class CodeMirrorSearchHighlighter {
                     m.position,
                     m.position + m.text.length
                   )
-                )
+                ),
+                filter: () => false
```

Comment:
It seems that this is the way: see for example the [match decorator](https://github.com/codemirror/view/blob/91e0a7d1e2c0fc1f7c57a9a29233128b617f1c1a/src/matchdecorator.ts#L143).

I guess the reason for the filtering is to be able to handle just the part of the document that is in the view port.

Reply:
> It seems that this is the way: see for example the [match decorator](https://github.com/codemirror/view/blob/91e0a7d1e2c0fc1f7c57a9a29233128b617f1c1a/src/matchdecorator.ts#L143)..
> 
> I guess the reason for the filtering is to be able to handle just the part of the document that is in the view port.

So this was added to fix https://github.com/jupyterlab/jupyterlab/issues/13885. FYI it only applies to old ranges, not the once that get added; the docs for `filter()` says:

> Filter the ranges already in the set. Only those for which this function returns true are kept.

Probably worth adding a comment.